File: src/main/java/org/sandboxpowered/silica/client/vulkan/error/VkError.java
package org.sandboxpowered.silica.client.vulkan.error;

import static org.lwjgl.vulkan.VK10.VK_SUCCESS;

public class VkError {
    public static void validate(int res, String message) {
        if (res == VK_SUCCESS)
            return;
        throw new RuntimeException(message);
    }
}

File: client/vulkan/src/main/kotlin/org/sandboxpowered/silica/client/vulkan/VkError.kt
package org.sandboxpowered.silica.client.vulkan

import org.lwjgl.vulkan.VK10

class VkError(message: String) : RuntimeException(message) {
    companion object {
        fun checkError(error: String, value: Int) {
            checkError(error, value, VK10.VK_SUCCESS)
        }

        fun <T> checkError(error: String, value: T, expectedValue: T) {
            if (value != expectedValue) {
                throw VkError(error)
            }
        }

        fun <T> checkError(error: String, value: T?, filter: (T?) -> Boolean) {
            if (filter(value)) {
                throw VkError(error)
            }
        }

        fun checkErrorRun(error: String, value: Int, onError: (Int) -> Unit) {
            if (value != VK10.VK_SUCCESS) {
                onError(value)
                throw VkError(error)
            }
        }
    }
}